.github/workflows/mergability_check.yml

```diff
@@ -0,0 +1,41 @@
+name: Validate and check if PR is mergable
+
+on:
+  pull_request:
+
+jobs:
+  do_merge_dry_run:
+    name: mergability_check_${{ github.event.pull_request.number }}
+    runs-on: linux.20_04.4x
+    env:
+        GH_RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+    steps:
+      - name: Checkout repo
+        id: checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+          check-latest: false
+          cache: pip
+          architecture: x64
+      - run: pip install pyyaml==6.0 rockset==1.0.3
+
+      - name: Dry Run Merge PR
+        if: github.event_name == 'pull_request'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Does the job run on pull request have access to these tokens?

Reply:
I believe so, it's at least done in upload-test-stats.yml